### PR TITLE
Link to user edit pages in importer messages

### LINKED ISSUE #1679: Link to user edit pages in importer messages
When the importer shows messages about users (such as "This new user has the same email as this existing user"), the respective users' edit pages should be linked in the message to enable managers to easily open the user profile for inspection.

### PR DESCRIPTION
**Pull request by jb3rndt**
Closes #1679

Preview:
grafik

First I had `user_edit_link` in the importer but then recognised that it is a pain forming the strings in the tests, so I moved it into tools. But im not sure if that is the right place for it

### CODE REVIEW

File: evap/staff/tools.py [hunk 1]
```diff
@@ -368,3 +369,11 @@ def remove_user_from_represented_and_ccing_users(user, ignored_users=None, test_
             cc_user.cc_users.remove(user)
             remove_messages.append(_("Removed {} from the CC users of {}.").format(user.full_name, cc_user.full_name))
     return remove_messages
+
+
+def user_edit_link(user_id):
+    return format_html(
+        '<a href="{}" target=_blank>edit user <span class="fas fa-external-link-alt"></span></a>'.format(
```
Comment (janno42) [inline evap/staff/tools.py:376]:
```suggestion
        '<a href="{}" target=_blank><span class="fas fa-user-pen"> edit user</span></a>'.format(
```

(will work only after #1733 is merged)

**Review by janno42 — COMMENTED** [review]
✔️ Meets requirements
✔️ UI functionality checked

File: evap/staff/tests/test_views.py [hunk 2]
```diff
@@ -489,8 +490,8 @@ def test_warning_handling(self):
         self.assertContains(
             reply,
             "The existing user would be overwritten with the following data:<br />"
-            " -  None None, lucilia.manilium@institution.example.com (existing)<br />"
-            " -  Lucilia Manilium, lucilia.manilium@institution.example.com (new)",
+            " -  None None, lucilia.manilium@institution.example.com (existing) [{}]<br />"
+            " -  Lucilia Manilium, lucilia.manilium@institution.example.com (new)".format(user_edit_link(user.pk)),
```
Comment (He3lixxx) [inline evap/staff/tests/test_views.py:494]: Could you please add a unit test for user_edit_link? Currently, all tests just test that method against itself, so we wouldn't notice if it was broken.

File: evap/staff/tools.py [hunk 3]
```diff
@@ -368,3 +369,11 @@ def remove_user_from_represented_and_ccing_users(user, ignored_users=None, test_
...
+            reverse("staff:user_edit", kwargs={"user_id": user_id})
+        )
+    )
```
Comment (He3lixxx) [inline evap/staff/tools.py:379]:
```suggestion
    return format_html(
        '<a href="{}" target=_blank><span class="fas fa-user-pen"> {}</span></a>',
        reverse("staff:user_edit", kwargs={"user_id": user_id}),
        _("edit user"),
    )
```

**Review by He3lixxx — CHANGES_REQUESTED** [review]
Coverage reporting seems off -- will have to look at it when this is rebased properly.

File: evap/staff/tools.py [hunk 4]
```diff
@@ -368,3 +369,12 @@ def remove_user_from_represented_and_ccing_users(user, ignored_users=None, test_
...
     return remove_messages
+
+
+def user_edit_link(user_id):
+    return format_html(
+        '<a href="{}" target=_blank><span class="fas fa-user-pen"> {}</span></a>'.format(
+            reverse("staff:user_edit", kwargs={"user_id": user_id}),
+            _("edit user"),
+        )
+    )
```
Comment (He3lixxx) [inline evap/staff/tools.py:380]: This `.format`-inside-`format_html` construction will "abuse" `format_html` as `mark_safe`. Both format arguments should be properly escaped.

```suggestion
    return format_html(
        '<a href="{}" target=_blank><span class="fas fa-user-pen"> {}</span></a>',
        reverse("staff:user_edit", kwargs={"user_id": user_id}),
        _("edit user"),
    )
```

**Review by He3lixxx — CHANGES_REQUESTED** [review]

File: evap/staff/importers.py [hunk 5]
```diff
@@ -326,7 +326,12 @@ def _create_user_data_mismatch_warning(user, user_data, test_run):
...
             msg
-            + format_html("<br /> - {} ({})", ExcelImporter._create_user_string(user), _("existing"))
+            + format_html(
+                "<br /> - {} ({}) [{}]",
+                ExcelImporter._create_user_string(user),
+                _("existing"),
+                user_edit_link(user.pk),
+            )
             + format_html("<br /> - {} ({})", ExcelImporter._create_user_string(user_data), _("new"))
         )
```
Comment (niklasmohrin) [inline evap/staff/importers.py:336]: I think this is equivalent to
```python
return format_html(
    "{}<br /> - {} ({}) [{}]<br /> - {} ({})",
    msg,
    ExcelImporter._create_user_string(user),
    _("existing"),
    user_edit_link(user.pk),
    ExcelImporter._create_user_string(user_data),
    _("new"),
)
```
Furthermore, you could remove the single argument `format_html` with `msg` above

Comment (He3lixxx) [reply evap/staff/importers.py:336]: Ahh yes good old `{}<br /> - {} ({}) [{}]<br /> - {} ({})`

**Review by He3lixxx — APPROVED** [review]

File: evap/staff/importers.py (hunk 5 above)
Comment (jb3rndt) [reply evap/staff/importers.py:336]: Actually true ^^

File: evap/staff/tools.py [hunk 6]
```diff
@@ -368,3 +369,11 @@ def remove_user_from_represented_and_ccing_users(user, ignored_users=None, test_
             cc_user.cc_users.remove(user)
             remove_messages.append(_("Removed {} from the CC users of {}.").format(user.full_name, cc_user.full_name))
     return remove_messages
+
+
+def user_edit_link(user_id):
+    return format_html(
+        '<a href="{}" target=_blank><span class="fas fa-user-pen"> {}</span></a>',
```
Comment (janno42) [inline evap/staff/tools.py:376]:
```suggestion
        '<a href="{}" target=_blank><span class="fas fa-user-pen"></span> {}</a>',
```
i'm sorry, my fault

**Review by janno42 — CHANGES_REQUESTED** [review]

File: evap/staff/importers.py [hunk 7]
```diff
@@ -324,30 +324,41 @@ def _create_user_data_mismatch_warning(user, user_data, test_run):
...
 
         return format_html(
-            _("The following user was previously marked inactive and is now marked active upon importing: {}"),
+            _("The following user was previously marked inactive and is now marked active upon importing: {} [{}]"),
             user_string,
+            user_edit_link(user.pk),
         )
 
     def _create_user_name_collision_warning(self, user_data, users_with_same_names):
         warningstring = format_html(_("An existing user has the same first and last name as a new user:"))
```
Comment (niklasmohrin) [inline evap/staff/importers.py:354]: Could you change this single-argument `format_html` to be a `mark_safe`?

**Review by niklasmohrin — COMMENTED** [review]
Code looks good, seems to work too, but I haven't manually tested in depth, I leave that to @janno42 :)

**Review by janno42 — APPROVED** [review]
✔️ Meets requirements
✔️ UI functionality checked

**Review by niklasmohrin — APPROVED** [review]

**Pull request merged**